pkg/util/net.go

```diff
@@ -34,16 +34,33 @@ import (
 
 var previousGatewayIP string = ""
 
-func GetHostIP() (string, error) {
-	ip, err := net.ChooseHostInterface()
-	if err == nil {
-		return ip.String(), nil
+func GetHostIP(nodeIP string) (string, error) {
+	var hostIP string
+	var err error
+
+	if nodeIP != "" {
+		klog.Infof("try to find nodeIP %s on host", nodeIP)
+		hostIP, err = selectV4IPFromHostInterface(nodeIP)
+		if err != nil {
+			return "", fmt.Errorf("failed to find nodeIP %s on host: %v", nodeIP, err)
+		}
+		goto found
 	}
-	hostIP := ip.String()
+
+	if ip, err := net.ChooseHostInterface(); err == nil {
+		hostIP = ip.String()
+	} else {
+		klog.Infof("failed to get host IP by default route: %v", err)
+		if hostIP, err = selectV4IPFromHostInterface(""); err != nil {
+			return "", err
+		}
+	}
+
+found:
 	if hostIP != previousGatewayIP {
 		previousGatewayIP = hostIP
-		klog.V(2).Infof("host gateway IP address: %s", hostIP)
 	}
+	klog.V(2).Infof("host gateway IP address: %s", hostIP)
```

Comment:
I think we moved this line up inside the `if` block where it was because we saw the same message, with the same IP, every time the check ran and it made the logs very noisy. I feel like we had a bug ticket for that change a while back.